c7n/resources/awslambda.py

```diff
@@ -326,7 +326,7 @@ def format_resource(self, r):
              'DeadLetterConfig',
              'Environment',
              'Handler',
-             'KMSKeyArn',
+             'KmsKeyArn',
```

Comment:
We can't include `KmsKeyArn` in this list, because the Lambda function resource will have any KMS key listed in `KMSKeyArn` instead.

We _do_ need to rename that key in order to fix `post-finding` though. One way we might do that is:
* Remove `KMSKeyArn` from the list of keys here.
* Add a follow-up line like `details['KmsKeyArn'] = r.get('KMSKeyArn')` after the `select_keys()` call